### PR TITLE
Re-implement #295 against latest master

### PR DESCRIPTION
**Pull request by natke**
Re-implemented #295 rather than resolve merge conflicts.

### CODE REVIEW

File: articles/language/type-model.md [hunk 1]
```diff
@@ -472,3 +472,69 @@ a valid invocation of `Controlled Rz` (note the parentheses around `0.1, target`
...
+```
+
+This example of a function comes from the [PhaseEstimation](https://github.com/Microsoft/Quantum/tree/master/Samples/PhaseEstimation) sample. It contains purely classical code. You can see that, unlike the example above, no qubits are allocated, and no quantum operations are used.
+
+```qsharp
+/// # Summary
+/// Given two arrays, returns a new array that is the pointwise product
+/// of each of the given arrays.
+function MultiplyPointwise (left : Double[], right : Double[]) : Double[] {
+
```
Comment (cgranade) [inline articles/language/type-model.md]:
```suggestion
```

File: articles/language/type-model.md [hunk 2]
```diff
@@ -472,3 +472,69 @@ a valid invocation of `Controlled Rz` (note the parentheses around `0.1, target`
...
+    mutable product = new Double[Length(left)];
+
+    for (idxElement in 0 .. Length(left) - 1) {
```
Comment (cgranade) [inline articles/language/type-model.md]:
```suggestion
    for (idxElement in IndexRange(left)) {
```

File: articles/language/type-model.md [hunk 3]
```diff
@@ -472,3 +472,69 @@ a valid invocation of `Controlled Rz` (note the parentheses around `0.1, target`
...
+        set product[idxElement] = left[idxElement] * right[idxElement];
```
Comment (cgranade) [inline articles/language/type-model.md]:
```suggestion
        set product w/= idxElement <- left[idxElement] * right[idxElement];
```

File: articles/language/type-model.md [hunk 4]
```diff
@@ -472,3 +472,69 @@ a valid invocation of `Controlled Rz` (note the parentheses around `0.1, target`
...
+    let n = Length(qubits);
+
+    for (i in 0 .. Length(masks) - 1) {
+        let (controls, targets) = (masks[i])!;
+        let controlBits = IntegerBits(controls, n);
+        let targetBits = IntegerBits(targets, n);
+        let cQubits = Subarray(controlBits, qubits);
+        let tQubits = Subarray(targetBits, qubits);
+
+        for (t in 0 .. Length(tQubits) - 1) {
```
Comment (cgranade) [inline articles/language/type-model.md]:
```suggestion
        for (target in tQubits) {
```

File: articles/language/type-model.md [hunk 5]
```diff
@@ -472,3 +472,69 @@ a valid invocation of `Controlled Rz` (note the parentheses around `0.1, target`
...
+            set result = result + [MCTGate(cQubits, tQubits[t])];
```
Comment (cgranade) [inline articles/language/type-model.md]:
```suggestion
            set result += [MCTGate(cQubits, target)];
```

File: articles/language/type-model.md [hunk 6]
```diff
@@ -472,3 +472,69 @@ a valid invocation of `Controlled Rz` (note the parentheses around `0.1, target`
...
+
+### Examples
+
+This example of a Q# operation comes from the [Measurement](https://github.com/Microsoft/Quantum/tree/master/Samples/Measurement) sample. Within operations we can allocate qubits and use quantum operations on those qubits such as `H` and `X`:
+
+```qsharp
+/// # Summary
+/// Creates a state and measures it in the Pauli-Z basis.
+operation MeasurementOneQubit () : Result {
+        mutable result = Zero;
```
[the remark this answers is not in the document]
Comment (natke) [reply articles/language/type-model.md]: Fixed

File: articles/language/type-model.md [hunk 7]
```diff
@@ -472,3 +472,69 @@ a valid invocation of `Controlled Rz` (note the parentheses around `0.1, target`
...
+
+        using (qubits = Qubit[1]) { // Allocate a qubit
+            let qubit = qubits[0];
+            H(qubit);               // Use a quantum operation on that qubit
+
```
[the remark this answers is not in the document]
Comment (natke) [reply articles/language/type-model.md]: Fixed

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:

| Article | Acrolinx<br>Score | Lowest<br>Category<br>Score | Issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/language/type-model.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/ba8c5289ace10582a23d158c05a6a9a88fa6ce2b/articles/language/type-model.md) | 79 | 56 | 111 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/9e6dc20e-ab39-4932-b705-4b0a441834e2_report.html) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/en-us/help/contribute/contribute-acrolinx-vscode?branch=master)
- [False positives or issues](https://aka.ms/acrolinxbug)

File: articles/language/type-model.md [hunk 8]
```diff
@@ -472,3 +472,70 @@ a valid invocation of `Controlled Rz` (note the parentheses around `0.1, target`
...
+### Examples
+
+This example of a Q# operation comes from the [Measurement](https://github.com/Microsoft/Quantum/tree/master/Samples/Measurement) sample. Within operations, we can allocate qubits and use quantum operations on those qubits such as `H` and `X`:
+
+```qsharp
+/// # Summary
+/// Prepares a state and measures it in the Pauli-Z basis.
+operation MeasureOneQubit () : Result {
+        result = Zero;
```
Comment (cgranade) [inline articles/language/type-model.md]: I think we still need the `mutable` keyword here to make the declaration that's used in 489.
```suggestion
        mutable result = Zero;
```

File: articles/language/type-model.md [hunk 9]
```diff
@@ -472,3 +472,70 @@ a valid invocation of `Controlled Rz` (note the parentheses around `0.1, target`
...
+
+        using (qubit = Qubit()) { // Allocate a qubit
+            H(qubit);               // Use a quantum operation on that qubit
+
+            result = M(qubit);      // Measure the qubit
```
Comment (cgranade) [inline articles/language/type-model.md]: We need to use the `set` keyword here to update the mutable binding.
```suggestion
            set result = M(qubit);      // Measure the qubit
```

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:

| Article | Acrolinx<br>Score | Lowest<br>Category<br>Score | Issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/language/type-model.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/0781c88077d28a5381bc8ebd5728ddad75b75962/articles/language/type-model.md) | 79 | 56 | 111 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/d3b38340-dbdd-44e3-bdcc-ad4e5a53695c_report.html) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/en-us/help/contribute/contribute-acrolinx-vscode?branch=master)
- [False positives or issues](https://aka.ms/acrolinxbug)

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:

| Article | Acrolinx<br>Score | Lowest<br>Category<br>Score | Issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/language/type-model.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/ecd219a7fca43b0f5539383271ff47be19fc6ad8/articles/language/type-model.md) | 79 | 56 | 111 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/bca30b56-8a78-4d65-ab68-4a34abd9ccc1_report.html) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/en-us/help/contribute/contribute-acrolinx-vscode?branch=master)
- [False positives or issues](https://aka.ms/acrolinxbug)

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:

| Article | Acrolinx<br>Score | Lowest<br>Category<br>Score | Issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/language/type-model.md](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/ecd219a7fca43b0f5539383271ff47be19fc6ad8/articles/language/type-model.md) | 79 | 56 | 111 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/bfae06cf-0375-4a29-8619-643f42f24e0d_report.html) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/en-us/help/contribute/contribute-acrolinx-vscode?branch=master)
- [False positives or issues](https://aka.ms/acrolinxbug)

**Pull request merged**